Implement overview-styling with grid

norbye opened the pull request:
# Description

Implemented the styling of the frontpage through CSS grid. The goal of this implementation is twofold;

1.  The overview both looks like a grid and smells like a grid - so imo it should be a grid
2. Plus this separates the components from the styling between them - facilitating moving components around and replacing them with improved versions without affecting the rest of the page.

Removed the two containers used for desktop and mobile and rather merged them into one responsive container.

Also updated some of the styling to add a little bit more space to the page, and make the spacing more consistent (e.g. adding padding to the right side of the screen and not only the left).

# Result

If you've made visual changes, please check the boxes below and include images showing the changes. Descriptions are appreciated.

- [x] Changes look good on both light and dark theme.
- [x] Changes look good with different viewports (mobile, tablet, etc.).
- [x] Changes look good with slower Internet connections.

> [!CAUTION]
> Make sure your images do not contain any real user information.

## Before

**NOTE: what looks like a right-overflow in the video only appears when you change the viewport. When you keep the viewport the same, it fills the screen**

https://github.com/webkom/lego-webapp/assets/13599770/68f552f7-8087-4565-b4df-1dbd50d3adc4

## After

https://github.com/webkom/lego-webapp/assets/13599770/4672269c-679b-4781-b242-4243f6b3acca

# Testing

- [x] I have thoroughly tested my changes.

Please describe what and how the changes have been tested, and provide instructions to reproduce if necessary.

---

Resolves ABA-832, ABA-423, ABA-727

ivarnakken reviewed (approved):
Much much cleaner! 😍 

This definitely "smells like grid"!

itsisak reviewed (approved):
Smart!

haukkagu reviewed (commented):
I love it! The only thing I slightly dislike is this empty space that occurs when the poll-component is too short:
![image](https://github.com/webkom/lego-webapp/assets/66320400/b480a69b-0167-43f7-a79f-8a760ef59016)

ivarnakken commented:
> The only thing I slightly dislike is this empty space that occurs when the poll-component is too short:

I agree, but I think we'll redesign this page quite soon so it won't really be a problem for long.

Nevertheless, now that this is refactored I believe it's easier to "fix this" if we want to.

norbye commented:
> I love it! The only thing I slightly dislike is this empty space that occurs when the poll-component is too short:

Absolutely agree. I've been holding on merging this as I was thinking about replacing the poll component with the weekly component - as that should always be the same height as the pinned article component.

Or I might make the left section into a sub grid to enable the right side to resize on the y-axis independently of the left side - but I'm more keen on promoting the weekly-section as a first iteration

norbye commented:
Moved the weekly component up to pinned articles now, and updated the styling to match the height of pinned articles.

Tested with the longest length a description can have, so this should keep the styling looking good regardless - with the presumption that they will continue keeping the title in the same format as it is now.

Also the padding around the image annoyed me, so I removed it(:

<img width="988" alt="image" src="https://github.com/webkom/lego-webapp/assets/13599770/f00bb01d-dad4-4847-bf1f-77c08697d4b1">

haukkagu reviewed (commented):
Wow! That looks a lot cleaner 😍 Very good idea!!

Another thing I noticed was that the readme component becomes very long:
![image](https://github.com/webkom/lego-webapp/assets/66320400/2408d45f-88ee-4e0d-8913-c73ff11967f5)

norbye commented:
> Another thing I noticed was that the readme component becomes very long:

Ah yeah, that component has been poorly styled for a while - so that happens on a few screen sizes already, but it would appear to happen more frequently now that I increased the padding and gave the component less space.

Will see what I can do about it

norbye commented:
Updated the styling of the latestReadme component now so that it's always displayed as a two column grid.

Also found out that the public frontpage never fetches readme's, so thats just an empty component rn - which is now fixed.

Regarding ABA-423, I kept only two readmes on the public frontpage, as the neighbor component looks weird with 4 as the height difference is too big.

Resolves ABA-423, ABA-727

Now there shouldn't be any more minor issues with the frontpage - but if there is I think it's time to look at that in another PR (unless it's caused by this PR and looks horrible 🤷 )

ivarnakken commented on packages/lego-bricks/src/components/Layout/Flex.tsx:
I don't see a problem with this - but is it used?

norbye replied:
> Edit: Oh and the componentRef attribute is something I needed while the styling was still using the Flex component - but it's something I've thought about having a use for in other PRs I've done so I want to keep it even though I ended up not using it

ivarnakken reviewed (approved):
Beautiful!!

norbye commented:
> Sort of related to this; why do we even separate this? Why not just show the exact same component with 4 readmes? I suppose it's nice to show the component expanded on the public frontpage to showcase it clearer - but why not show 4 there as well?

I kept it that way because the extra height of the readmes affected the spacing on the rest of the page - and I didn't really want to have the extra space under the pinned thingy;

With two readmes;
<img width="1144" alt="image" src="https://github.com/webkom/lego-webapp/assets/13599770/392798bc-6521-4cc9-a13e-cb195e604cd1">


With four readmes;
<img width="1154" alt="image" src="https://github.com/webkom/lego-webapp/assets/13599770/54f8fe00-c1b5-4804-8fa6-f3eeda18fc19">

So imo the 4-readmes work smoother on the logged in overview, as it only affects the components underneath, and not the components on its side
